pkg/authorization/decorator.go

```diff
@@ -42,24 +42,27 @@ type Decorator struct {
 	key    string
 }
 
-func NewDecorator(target authorizer.Authorizer, key string) *Decorator {
+// NewDecorator returns a new authorizer which is associated with the given key.
+// The prefix key must not contain a trailing slash `/`.
+// Decorating functions are applied in the order they have been invoked.
+func NewDecorator(key string, target authorizer.Authorizer) *Decorator {
```

Comment:
nit: when a function panics then we usually name it `NewDecoratorOrDie`

Reply:
indeed, good suggestion 👍